Number_types/include/CGAL/Number_types/internal/Exact_type_selector.h

```diff
@@ -93,8 +93,12 @@ struct Exact_ring_selector : Exact_field_selector < T > { };
 
 template <>
 struct Exact_ring_selector<double>
-#ifdef CGAL_HAS_MPZF
+#if BOOST_VERSION > 107900 && defined(CGAL_USE_BOOST_MP)
+{ typedef boost::multiprecision::cpp_int Type; };
```

Comment:
What are you trying to achieve? `Exact_ring_selector<double>` should provide a ring type **that contains `double`**, which is not the case of `cpp_int`.

Reply:
ah right I forget that point. Since Mpzf depends on GMP I was trying to not use it in case boost mp built-in was the "default" exact NT.

Reply:
If you want to centralize the logic in this file, we could make Exact_integer an alias for `Exact_ring_selector<int>` and move all the ifdefs from Exact_integer.h to Exact_type_selector.h. But that doesn't seem to be the goal here.
MP_Float is our only option without GMP, for `Exact_ring_selector<double>`, otherwise it will fall back to a rational type. But if you don't want GMP, why not tell it to CGAL with CGAL_NO_GMP?